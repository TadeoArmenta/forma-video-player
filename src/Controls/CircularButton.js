import React, {Component} from 'react';
import {
  TouchableHighlight,
  StyleSheet,
  Platform,
  TouchableNativeFeedback,
  View,
  Text
} from 'react-native';

export default class CircularButton extends Component {

  render() {

    const underlayColor = this.props.underlayColor || '#fff';

    const dimensions = {
      width: this.props.radius * 2,
      height: this.props.radius * 2,
    };

    if(Platform.OS === 'ios') {
      return (
        <TouchableHighlight
          onPress={this.props.onPress}
          style={[styles.commonViewStyle, this.props.style, dimensions]}
          underlayColor={underlayColor}
        >
          {this.props.children}
        </TouchableHighlight>
      )
    }
    else {
      return (
        <View style={[styles.commonViewStyle, this.props.style, dimensions]}>
          <TouchableNativeFeedback
            background={TouchableNativeFeedback.Ripple(underlayColor, true)}
            onPress={this.props.onPress}
          >
            <View style={[styles.commonViewStyle, dimensions]}>
              {this.props.children}
            </View>
          </TouchableNativeFeedback>
        </View>
      )
    }
  }

}

const styles = StyleSheet.create({
  commonViewStyle: {
    borderRadius: 50,
    alignItems: 'center',
    justifyContent: 'center'
  }
})
