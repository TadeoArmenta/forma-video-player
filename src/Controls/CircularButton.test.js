import React from 'react'
import CircularButton from './CircularButton'
import ShallowTestRenderer from 'react-test-renderer/shallow'

describe('CircularButton', () => {

  let renderer = new ShallowTestRenderer()
  
  it('should match snapshot', () => {
    const rendered = renderer.render(<CircularButton />)

    expect(rendered).toMatchSnapshot()
  })

})
