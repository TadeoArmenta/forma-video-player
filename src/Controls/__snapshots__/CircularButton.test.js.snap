// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularButton should match snapshot 1`] = `
<TouchableHighlight
  activeOpacity={0.85}
  onPress={undefined}
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "justifyContent": "center",
      },
      undefined,
      Object {
        "height": NaN,
        "width": NaN,
      },
    ]
  }
  underlayColor="#fff"
/>
`;
